terraform/core/46-mwaa-bucket-kms.tf

```diff
@@ -0,0 +1,88 @@
+data "aws_caller_identity" "current" {}
+
+locals {
+  custom_key_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Sid    = "Allow CloudWatch Logs to use the KMS key"
+        Effect = "Allow"
+        Principal = {
+          Service = "logs.amazonaws.com"
+        }
+        Action = [
+          "kms:Encrypt",
+          "kms:Decrypt",
+          "kms:ReEncrypt*",
+          "kms:GenerateDataKey*",
+          "kms:DescribeKey"
+        ]
+        Resource = "*"
+      },
+      {
+        Sid    = ""
+        Effect = "Allow"
+        Principal = {
+          AWS = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+        }
+        Action   = "kms:*"
+        Resource = "*"
```

Comment:
This allows all actions on the key

Reply:
I only allow Logs to use the KMS key. The other two sections are used on all the keys in our AWS account. Could we keep it as same as other now. What do you think?
You can check all the keys for example in the **DataPlatform-Pre-Production** account via below URL?
https://eu-west-2.console.aws.amazon.com/kms/home?region=eu-west-2#/kms/keys